pkg/sysfs/system.go

```diff
@@ -913,15 +922,17 @@ func (sys *system) discoverCPU(path string) error {
 		return fmt.Errorf("exactly one node per cpu allowed")
 	}
 
-	if sys.threads < 1 {
-		sys.threads = 1
-	}
-	if cpu.threads.Size() > sys.threads {
-		sys.threads = cpu.threads.Size()
-	}
-
 	sys.cpus[cpu.id] = cpu
 
+	if threadCnt := cpu.threads.Size(); threadCnt > 0 {
+		if threadCnt < sys.minThreads {
+			sys.minThreads = threadCnt
+		}
```

Comment:
If sys.minThreads is initialized to 1, then only a threadCnt of zero or less than zero will update its value. If the minimal threadCnt over all cpu.threads.Size() is for example 2, is it still ok to have min.Threads as 1?

Reply:
Thanks for spotting that. Updated.